### ISSUE TITLE
Lua Error on Opening Profession

### ISSUE THREAD

**Issue by LiangYuxuan** [Xruptor/BagSync#188]
BagSync version: v17.0
WoW version: 10.0.0.46340

Get following lua error on opening profession frame
```
2x Interface/AddOns/BagSync/wireframe/scanner.lua:400: attempt to call field 'GetTradeSkillLine' (a nil value)
[string "@Interface/AddOns/BagSync/wireframe/scanner.lua"]:400: in function `SaveProfessions'
[string "@Interface/AddOns/BagSync/wireframe/events.lua"]:318: in function `?'
[string "@Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:119: in function <...ings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:119>
[string "=[C]"]: ?
[string "@Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:29: in function <...ings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:25>
[string "@Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:64: in function `Fire'
[string "@Interface/AddOns/ElvUI/Libraries/Core/Ace/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <...vUI/Libraries/Core/Ace/AceEvent-3.0/AceEvent-3.0.lua:119>
```

And this one when hover on item icon in profession frame
```
3x Interface/AddOns/BagSync/wireframe/tooltip.lua:655: attempt to call field 'GetRecipeReagentItemLink' (a nil value)
[string "@Interface/AddOns/BagSync/wireframe/tooltip.lua"]:655: in function <Interface/AddOns/BagSync/wireframe/tooltip.lua:653>
[string "=[C]"]: in function `SetRecipeReagentItem'
[string "@Interface/AddOns/Blizzard_ProfessionsTemplates/Blizzard_ProfessionsRecipeSchematicForm.lua"]:715: in function <...emplates/Blizzard_ProfessionsRecipeSchematicForm.lua:713>

Locals:
self = GameTooltip {
 0 = <userdata>
 SetCurrencyByID = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 ItemTooltip = Frame {
 }
 SetCurrencyToken = <function> defined =[C]:-1
 hasMoney = 1
 textRight2Font = "GameTooltipText"
 ClearLines = <function> defined =[C]:-1
 Show = <function> defined =[C]:-1
 SetUnitDebuff = <function> defined =[C]:-1
 SetLFGDungeonReward = <function> defined @Interface/AddOns/AllTheThings/AllTheThings.lua:7592
 SetQuestItem = <function> defined =[C]:-1
 AttachComplete = true
 waitingForData = false
 SetToyByItemID = <function> defined =[C]:-1
 numMoneyFrames = 2
 comparing = false
 SetRecipeReagentItem = <function> defined =[C]:-1
 layoutType = "TooltipDefaultLayout"
 shoppingTooltips = <table> {
 }
 __tooltipUpdated = false
 SetLFGDungeonShortageReward = <function> defined @Interface/AddOns/AllTheThings/AllTheThings.lua:7609
 SetQuestLogItem = <function> defined =[C]:-1
 SetBackpackToken = <function> defined =[C]:-1
 BottomOverlay = Texture {
 }
 textRight1Font = "GameTooltipHeaderText"
 StatusBar = GameTooltipStatusBar {
 }
 NineSlice = Frame {
 }
 textLeft2Font = "GameTooltipText"
 TextRight2 = GameTooltipTextRight2 {
 }
 TextLeft2 = GameTooltipTextLeft2 {
 }
 SetUnitAura = <function> defined =[C]:-1
 SetMerchantCostItem = <function> defined =[C]:-1
 SetCurrencyTokenByID = <function> defined =[C]:-1
 SetUnitBuff = <function> defined =[C]:-1
 needsReset = false
 updateTooltipTimer = 0.012000
 TextLeft1 = GameTooltipTextLeft1 {
 }
 TextRight1 = GameTooltipTextRight1 {
 }
 TopOverlay = Texture {
 }
}
recipeID = 324088
reagentIndex = 1
(*temporary) = nil
(*temporary) = 324088
(*temporary) = 1
(*temporary) = "attempt to call field 'GetRecipeReagentItemLink' (a nil value)"
Tooltip = <table> {
 ItemCount = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:297
 HexColor = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:40
 Enable = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:290
 NewModule = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:231
 EnableModule = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:328
 modules = <table> {
 }
 GetModule = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:206
 IterateEmbeds = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:438
 defaultModuleLibraries = <table> {
 }
 UnitTotals = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:254
 SetDefaultModulePrototype = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:405
 name = "BagSync_Tooltip"
 IsEnabled = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:447
 orderedModules = <table> {
 }
 DisableModule = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:346
 ColorizeUnit = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:60
 TallyUnits = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:315
 GetSortIndex = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:47
 SetDefaultModuleState = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:383
 HookTooltip = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:576
 SetEnabledState = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:420
 enabledState = true
 __lastTally = <table> {
 }
 baseName = "BagSync"
 OnEnable = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:803
 HookBattlePetTooltip = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:760
 SetDefaultModuleLibraries = <function> defined @Interface/AddOns/ElvUI/Libraries/Core/Ace/AceAddon-3.0/AceAddon-3.0.lua:363
 CurrencyTooltip = <function> defined @Interface/AddOns/BagSync/wireframe/tooltip.lua:523
 IterateModules = 
```

**Comment by FireyDragon44** [comment]
I as well get this error upon opening professions...

Message: Interface/AddOns/BagSync/wireframe/scanner.lua:400: attempt to call field 'GetTradeSkillLine' (a nil value)
Time: Fri Oct 28 01:44:29 2022
Count: 1
Stack: Interface/AddOns/BagSync/wireframe/scanner.lua:400: attempt to call field 'GetTradeSkillLine' (a nil value)
[string "=(tail call)"]: ?
[string "@Interface/AddOns/BagSync/wireframe/scanner.lua"]:400: in function `SaveProfessions'
[string "@Interface/AddOns/BagSync/wireframe/events.lua"]:318: in function `?'
[string "@Interface/AddOns/AdiBags/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua"]:119: in function <...ags/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:119>
[string "=[C]"]: ?
[string "@Interface/AddOns/AdiBags/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua"]:29: in function <...ags/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:25>
[string "@Interface/AddOns/AdiBags/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua"]:64: in function `Fire'
[string "@Interface/AddOns/BagSync/libs/AceEvent-3.0/AceEvent-3.0.lua"]:120: in function <...ce/AddOns/BagSync/libs/AceEvent-3.0/AceEvent-3.0.lua:119>

Locals: <none>

**Comment by Xruptor** [comment]
Hmmm looks like I may or may not be able to fix this until 10.0.2.  They are changing how professions are interpreted in the game client.  Why am I not surprised Blizzard?  Break code every patch and expansion why don't you.

https://wowpedia.fandom.com/wiki/API_C_TradeSkillUI.GetProfessionInfoBySkillLineID

Let me guess it won't work in Classic/WOTLK because the code isn't compatible.  LOL I'll have to test it and see.
